internal/querycoordv2/meta/collection_manager.go

```diff
@@ -466,6 +392,55 @@ func (m *CollectionManager) putPartition(partitions []*Partition, withSave bool)
 	return nil
 }
 
+func (m *CollectionManager) UpdateLoadPercent(partitionID int64, loadPercent int32) (int32, error) {
+	m.rwmutex.Lock()
+	defer m.rwmutex.Unlock()
+
+	oldPartition, ok := m.partitions[partitionID]
+	if !ok {
+		return 0, merr.WrapErrPartitionNotFound(partitionID)
+	}
+
+	if loadPercent < oldPartition.LoadPercentage {
```

Comment:
may "<=" here? if call `putPartition` and `putColleciton` each check, load timeout mechanism will be broken since updatedAt will be modified even not progress changed